app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmTimePickerFragment.kt

```diff
@@ -0,0 +1,109 @@
+package nerd.tuxmobil.fahrplan.congress.alarms
+
+import nerd.tuxmobil.fahrplan.congress.R
+import nerd.tuxmobil.fahrplan.congress.extensions.getLayoutInflater
+import nerd.tuxmobil.fahrplan.congress.extensions.requireViewByIdCompat
+import nerd.tuxmobil.fahrplan.congress.BuildConfig
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+
+import android.os.Bundle
+import android.app.Dialog
+import android.view.View
+import android.widget.Spinner
+import android.widget.ArrayAdapter
+import android.content.DialogInterface
+import android.annotation.SuppressLint
+
+import androidx.annotation.MainThread
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentResultListener
+
+import java.lang.NullPointerException
+
+class AlarmTimePickerFragment : DialogFragment() {
+    private lateinit var spinner: Spinner
+    private var alarmTimeIndex = 0
+    @MainThread
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val context = requireContext()
+
+        alarmTimeIndex = AppRepository.readAlarmTimeIndex()
+
+        @SuppressLint("InflateParams")
+        val layout = context.getLayoutInflater().inflate(
+            R.layout.reminder_dialog,
+            null,
+            false
+        )
+        // https://possiblemobile.com/2013/05/layout-inflation-as-intended/
+
+        initializeSpinner(layout)
+
+        return AlertDialog.Builder(context).apply {
+            setView(layout)
+            setTitle(R.string.choose_alarm_time)
+            setPositiveButton(android.R.string.ok) {
+                _: DialogInterface?,
+                _: Int -> passBackAlarmTimesIndex()
+            }
+            setNegativeButton(android.R.string.cancel, null)
+        }.create()
+    }
+
+    private fun initializeSpinner(rootView: View) {
+        val arrayAdapter = ArrayAdapter.createFromResource(
+            rootView.context,
+            R.array.preference_entries_alarm_time_titles,
+            android.R.layout.simple_spinner_item
+        )
+        arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+
+        spinner = rootView.requireViewByIdCompat<Spinner>(R.id.spinner).apply{
+            adapter = arrayAdapter
+            setSelection(alarmTimeIndex)
+        }
+    }
+
+    private fun passBackAlarmTimesIndex() {
+        val alarmTimesIndex = spinner.selectedItemPosition
+
+        val requestKey = requireArguments().getString(REQUEST_KEY_BUNDLE_KEY)
+            ?: throw NullPointerException("Request key must be passed via fragment arguments.")
+
+        val result = Bundle().apply {
+            putInt(ALARM_TIMES_INDEX_BUNDLE_KEY, alarmTimesIndex)
+        }
+
+        parentFragmentManager.setFragmentResult(requestKey, result)
+    }
+
+    companion object {
```

Comment:
Please move the companion object up to the top because it contains the public `show` function which acts as the entry point for the class.